libs/wire-api/test/unit/Test/Wire/API/Roundtrip.hs

```diff
@@ -0,0 +1,320 @@
+      testRoundTrip @Team.Feature.SSOTeamConfig,
+      testRoundTrip @Team.Invitation.InvitationRequest,
+      testRoundTrip @Team.Invitation.Invitation,
+      testRoundTrip @Team.Invitation.InvitationList,
+      testRoundTrip @Team.LegalHold.NewLegalHoldService,
+      testRoundTrip @Team.LegalHold.ViewLegalHoldServiceInfo,
+      testRoundTrip @Team.LegalHold.NewLegalHoldService,
+      testRoundTrip @Team.LegalHold.ViewLegalHoldService,
+      testRoundTrip @Team.LegalHold.UserLegalHoldStatusResponse,
+      testRoundTrip @Team.LegalHold.RemoveLegalHoldSettingsRequest,
+      testRoundTrip @Team.LegalHold.DisableLegalHoldForUserRequest,
+      testRoundTrip @Team.LegalHold.ApproveLegalHoldForUserRequest,
+      testRoundTrip @Team.LegalHold.External.RequestNewLegalHoldClient,
+      testRoundTrip @Team.LegalHold.External.NewLegalHoldClient,
+      testRoundTrip @Team.LegalHold.External.LegalHoldServiceConfirm,
+      testRoundTrip @Team.LegalHold.External.LegalHoldServiceRemove,
+      testRoundTrip @Team.Member.TeamMember,
+      testRoundTrip @Team.Member.ListType,
+      testRoundTrip @Team.Member.TeamMemberList,
+      testRoundTrip @Team.Member.NewTeamMember,
+      testRoundTrip @Team.Member.TeamMemberDeleteData,
+      testRoundTrip @Team.Permission.Permissions,
+      testRoundTrip @Team.Role.Role,
+      testRoundTrip @Team.SearchVisibility.TeamSearchVisibility,
+      testRoundTrip @Team.SearchVisibility.TeamSearchVisibilityView,
+      testRoundTrip @Team.SearchVisibility.TeamSearchVisibilityAvailable,
+      testRoundTrip @Team.SearchVisibility.TeamSearchVisibilityAvailableView,
+      testRoundTrip @User.NewUser,
+      testRoundTrip @User.NewUserPublic,
+      testRoundTrip @User.UserIdList,
+      testRoundTrip @User.UserProfile,
+      testRoundTrip @User.User,
+      testRoundTrip @User.SelfProfile,
+      testRoundTrip @User.InvitationCode,
+      testRoundTrip @User.BindingNewTeamUser,
+      -- FUTUREWORK: this should probably be tested individually,
+      -- but NewUserOrigin doesn't have JSON instances, just plain functions.
+      -- testRoundTrip @User.NewUserOrigin,
+      testRoundTrip @User.UserUpdate,
+      testRoundTrip @User.PasswordChange,
+      testRoundTrip @User.LocaleUpdate,
+      testRoundTrip @User.EmailUpdate,
+      testRoundTrip @User.PhoneUpdate,
+      testRoundTrip @User.HandleUpdate,
+      testRoundTrip @User.DeleteUser,
+      testRoundTrip @User.VerifyDeleteUser,
+      testRoundTrip @User.DeletionCodeTimeout,
+      testRoundTrip @User.Activation.ActivationKey,
+      -- FUTUREWORK: this should probably be tested individually,
+      -- but ActivationTarget currently doesn't have JSON instances itself.
+      -- testRoundTrip @User.Activation.ActivationTarget,
+      testRoundTrip @User.Activation.ActivationCode,
+      testRoundTrip @User.Activation.Activate,
+      currentlyFailing (testRoundTrip @User.Activation.ActivationResponse), -- because ToJSON doesn't serialize 'sso_id'
```

Comment:
It might make sense to some degree, since SSO users don't need to activate their account (unless I'm mistaken). However, then the types should be adjusted to not allow this. I will leave a comment.

In the meantime, serializing this field (even though it should normally not occur) is probably better than silently dropping it, so I'll do that.